examples/e2etest.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use astarte_sdk::builder::AstarteOptions;
+use log::error;
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+
+    let realm = "test";
+    let device_id = std::env::var("E2E_DEVICE_ID").unwrap();
+    let credentials_secret = std::env::var("E2E_CREDENTIALS_SECRET").unwrap();
+    let pairing_url = "https://api.autotest.astarte-platform.org/pairing";
+
+    let sdk_options = AstarteOptions::new(realm, &device_id, &credentials_secret, pairing_url)
+        .interface_file(std::path::Path::new(
+            "./examples/interfaces/com.test.Everything.json",
+        ))
+        .unwrap()
+        .ignore_ssl_errors()
+        .build();
+
+    let mut device = astarte_sdk::AstarteSdk::new(&sdk_options).await.unwrap();
+
+    let w = device.clone();
+    tokio::task::spawn(async move {
+        for i in 0..3 {
+            let ret = w.send("com.test.Everything", "/integer", i).await;
+
+            if let Err(err) = ret {
+                error!("send error {:?}", err);
+                std::process::exit(1);
+            }
+
+            std::thread::sleep(std::time::Duration::from_millis(300));
+        }
+
+        std::thread::sleep(std::time::Duration::from_millis(1000));
+        println!("Test completed successfully");
```

Comment:
For the e2e test to be meaningful, one should also check that the value is correctly received from Astarte. This is important especially for datastream interfaces having  mappings with reliability `unreliable` (default). At the moment, we do not know if data have actually reached Astarte.

To retrieve data, you can use any HTTP client:
 - Generate the JWT for authorization using astartectl: `astartectl utils gen-jwt all-realm-apis`. 
 - Query AppEngine at `/{realm_name}/devices/{device_id}/interfaces/{interface}[/{path}]`, 
 - Parse the resulting JSON object and finally compare with expected values.

Reply:
Since the other pr is temporarily blocked, I have made another commit here with those checks, it is now ready for review (again)